cmd/multiclusterapp.go

```diff
@@ -0,0 +1,668 @@
+						Name:  "target,t",
+						Usage: "Target project ids to upgrade, leave it empty to keep current targets",
+					},
+				},
+			},
+			cli.Command{
+				Name:        "list-templates",
+				Aliases:     []string{"lt"},
+				Usage:       "List templates available for installation",
+				Description: "\nList all app templates in the current Rancher server",
+				ArgsUsage:   "None",
+				Action:      globalTemplateLs,
+				Flags: []cli.Flag{
+					formatFlag,
+					cli.StringFlag{
+						Name:  "catalog",
+						Usage: "Specify the catalog to list templates for",
+					},
+				},
+			},
+			cli.Command{
+				Name:        "show-template",
+				Aliases:     []string{"st"},
+				Usage:       "Show versions available to install for an app template",
+				Description: "\nShow all available versions of an app template",
+				ArgsUsage:   "[TEMPLATE_ID]",
+				Action:      templateShow,
+			},
+			cli.Command{
+				Name:      "show-app",
+				Aliases:   []string{"sa"},
+				Usage:     "Show an app's available versions and revisions",
+				ArgsUsage: "[APP_NAME/APP_ID]",
+				Action:    showMultiClusterApp,
+				Flags: []cli.Flag{
+					formatFlag,
+				},
+			},
+		},
+	}
+}
+
+func multiClusterAppLs(ctx *cli.Context) error {
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	collection, err := c.ManagementClient.MultiClusterApp.List(defaultListOpts(ctx))
+	writer := NewTableWriter([][]string{
+		{"NAME", "App.Name"},
+		{"STATE", "App.State"},
+		{"VERSION", "Version"},
+		{"PROJECT_TARGETS", "Targets"},
+	}, ctx)
+
+	defer writer.Close()
+
+	for _, item := range collection.Data {
+		templateVersion, err := c.ManagementClient.TemplateVersion.ByID(item.TemplateVersionID)
+		if err != nil {
+			return err
+		}
+		targetNames, err := getReadableTargetNames(c.ManagementClient, item.Targets)
+		if err != nil {
+			return err
+		}
+		writer.Write(&MultiClusterAppData{
+			App:     item,
+			Version: templateVersion.Version,
+			Targets: strings.Join(targetNames, ","),
+		})
+	}
+	return writer.Err()
+}
+
+func getReadableTargetNames(client *managementClient.Client, targets []managementClient.Target) ([]string, error) {
+	var targetNames []string
+	for _, target := range targets {
+		clusterID, _ := parse(target.ProjectID)
+		cluster, err := client.Cluster.ByID(clusterID)
+		if err != nil {
+			return nil, err
+		}
+		project, err := client.Project.ByID(target.ProjectID)
+		if err != nil {
+			return nil, err
+		}
+		targetNames = append(targetNames, fromStrings(cluster.Name, project.Name))
+	}
+	return targetNames, nil
+}
+
+func multiClusterAppDelete(ctx *cli.Context) error {
+	if ctx.NArg() == 0 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, arg := range ctx.Args() {
+		resource, err := Lookup(c, arg, managementClient.MultiClusterAppType)
+		if err != nil {
+			return err
+		}
+
+		app, err := c.ManagementClient.MultiClusterApp.ByID(resource.ID)
```

Comment:
I see that the other delete commend will do it in this way, but the `.Delete()` function only requires `type.Resource` in the object. Maybe we only need to wrap the `resource` returned by `Lookup` function instead of using `ByID` to get it.